### PR TITLE
Refactor: Cleanup CMake files

### PR DESCRIPTION
**Pull request by zaufi**
In this PR:

- streamline build system a little
- use some things right
- consistent style and formatting

Feel free to ask your "WHY?" questions if any ;-)

### CODE REVIEW

File: src/phase/network/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,13 @@
-add_library(network STATIC TcpSocket.cpp)
-target_include_directories(network INTERFACE "${CMAKE_CURRENT_LIST_DIR}/")
+add_library(
+    network
+    TcpSocket.cpp
+    TcpSocket.hpp
```
Comment (mutanabbi) [inline src/phase/network/CMakeLists.txt:4]: What is a reason to add header files to a library?

File: src/hierarchy/CMakeLists.txt [hunk 2]
```diff
@@ -25,12 +25,14 @@ add_library(
...
 
 target_link_libraries(
     battle
     PRIVATE
         bestiary
-)
+  )
 
-add_subdirectory(tests)
+if(BUILD_TESTING)
```
Comment (mutanabbi) [inline src/hierarchy/CMakeLists.txt:36]: Does it mean I need to use `-DBUILD_TESTING` in CI explicitly?

Comment (zaufi) [reply src/hierarchy/CMakeLists.txt:36]: No. CTest module calls `enable_testing()` by default, so it's `ON` by default. But when it's `OFF` better not to do things related to testing and save some configuring/building time.

File: src/phase/network/CMakeLists.txt (hunk 1 above)
Comment (zaufi) [reply src/phase/network/CMakeLists.txt:4]: There are a few of them:
* explicit better than implicit
* the main reason is to help CMake to track dependencies (and pass it to the underlaid build system), cuz not all generators support compiler-provided dependencies in form of GNU Make
* for IDE generators it brings the source to the list of target sources

Comment (mutanabbi) [reply src/phase/network/CMakeLists.txt:4]: What is your recommendation for `add_executable`? To list headers too as well as cpp's?

File: src/hierarchy/CMakeLists.txt (hunk 2 above)
Comment (mutanabbi) [reply src/hierarchy/CMakeLists.txt:36]: Are you sure? I've just tested
```
enable_testing()
if(BUILD_TESTING)
  add_subdirectory(tests)
endif()
```
```
> cmake -DCMAKE_BUILD_TYPE=Debug . 
> cmake --build .
```
it didn't create any unittests at all

File: src/phase/network/CMakeLists.txt (hunk 1 above)
Comment (zaufi) [reply src/phase/network/CMakeLists.txt:4]: Sure. The rules are the same to apply

File: src/hierarchy/CMakeLists.txt (hunk 2 above)
Comment (zaufi) [reply src/hierarchy/CMakeLists.txt:36]: Did u `include(CTest)` module? Please, read the docs on it.

**Pull request merged**